src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectSamplesPageIT.java

```diff
@@ -305,4 +308,67 @@ public void testAddNewSamples() {
 		page.enterSampleName("GOOD_NAME");
 		assertFalse(page.isSampleNameErrorDisplayed(), "Sample name error should not be displayed");
 	}
+
+	@Test
+	public void testFilterByFile() {
+		String invalidName = "name_not_in_list";
+		int numberValidSampleNames = 5;
+
+		LoginPage.loginAsManager(driver());
+		ProjectSamplesPage page = ProjectSamplesPage.gotToPage(driver(), 1);
+
+		TableSummary summary = page.getTableSummary();
+		assertEquals(23, summary.getTotal(), "Without the filter there should be 23 elements in the table");
+
+		page.filterByFile("src/test/resources/files/filter-by-file/sample-names.txt");
+		List<String> invalidSamples = page.getInvalidSampleNames();
+		assertEquals(1, invalidSamples.size(), "Should have one invalid sample");
+		assertEquals(invalidName, invalidSamples.get(0), "Should have the correct invalid sample");
+
+		page.submitFilterByFile();
+
+		summary = page.getTableSummary();
+		assertEquals(numberValidSampleNames, summary.getTotal(), "Should have the correct number of samples");
+	}
+
+	@Test
+	public void testFilterByFileWithAssociatedProjects() {
+		String ASSOCIATED_PROJECT_FILTER = "project6";
+		String ASSOCIATED_SAMPLE_NAME = "sample5fg44";
+
+		LoginPage.loginAsManager(driver());
+		ProjectSamplesPage page = ProjectSamplesPage.gotToPage(driver(), 1);
+
+		TableSummary summary = page.getTableSummary();
+		assertEquals(23, summary.getTotal(), "Without the filter there should be 23 elements in the table");
+
+		page.filterByFile("src/test/resources/files/filter-by-file/sample-names-with-associated.txt");
+		List<String> invalidSamples = page.getInvalidSampleNames();
+		invalidSamples.forEach(
+				name -> assertEquals(ASSOCIATED_SAMPLE_NAME, invalidSamples.get(0), "Should have the correct invalid sample"));
```

Comment:
I'm a little confused by this loop. You are looping through the invalidSamples but then only checking the first invalidSample that it matches.

Was this meant to be like line 372? or should it be like line 325 and line 326?